src/array.jl

```diff
@@ -212,6 +212,31 @@ end
 
             convert($A{T, N, R}, A)
         end
+
+        function Base.vcat{T,N,R}(A1::$A{T, N, R}, An::$A...)
+            As = (A1, An...)
+            levels = unique(T[[a.pool.levels for a in As]...;])
+            idx = [indexin(a.pool.index, levels) for a in As]
+
+            ordered = A1.pool.ordered
+            if ordered && levels != A1.pool.levels
+                warn("Failed to preserve order of levels. Define all levels in the first argument.")
```

Comment:
Good idea the check this, but we generally avoid in Julia printing warnings: either it should be an error, or it should work. Warnings that are printed when the code works are annoying for users.

I think it's fine to silently make the result unordered if the order cannot be respected. We'll have to document this at some point of course (for now the package has no docstrings).

We should also be able to make this more flexible: it's fine if levels are different as long as their orders are consistent. For example, `["a", "b", "c"]` and `["a", "b"]`/`["a", "b", "c", "d"]`/`["b", "c", "d"]` are fine. So you just need to check that levels for each input vectors are included (in the same order) in the output levels.

That requires a clever algorithm to compute the levels in the first place: from the `levels` computed above, which are sorted according to the rank of the input, I guess you could repeatedly sort it so that elements that are common with the levels of a given input are in the same order. If you do that for each input, you should get the correct order when possible, and a reasonable order else (in which case the check I described in the previous paragraph will fail, and an unordered result will be returned). This strategy could be applied to unordered inputs too.